### PR TITLE
[#356] Check response code, ensure required OpenID metadata given

### PR DESCRIPTION
**Pull request by MartinFlores751**
Using the same scenario given in issue:

```
$ irods_http_api config.json 
Validating configuration file ...
No JSON schema file provided. Using default.
Configuration passed validation!
[2024-11-05 20:21:04.138] [P:16496] [info] [T:16496] Initializing server.
[2024-11-05 20:21:04.138] [P:16496] [trace] [T:16496] Verifying OIDC endpoint configuration
[2024-11-05 20:21:04.140] [P:16496] [error] [T:16496] Did not get a status code 200 OK response from OpenID Provider.
[2024-11-05 20:21:04.140] [P:16496] [error] [T:16496] Invalid OIDC configuration, server not starting.
```

### CODE REVIEW

**Review by korydraughn — COMMENTED** [review]
Looks good.

Please update the commit message so it's clear what the change is addressing.

**Review by alanking — COMMENTED** [review]
I couldn't help it

File: core/src/main.cpp [hunk 1]
```diff
@@ -808,11 +808,25 @@ auto load_oidc_configuration(const json& _config, json& _oi_config, json& _endpo
 		// Sends and recieves response
```
Comment (alanking) [inline core/src/main.cpp:808]:
```suggestion
		// Sends and receives response
```

**Review by trel — APPROVED** [review]
seems good with alan's typo fix

**Comment by trel** [comment]
and agreed, better commit message please

File: core/src/main.cpp (hunk 1 above)
Comment (MartinFlores751) [reply core/src/main.cpp]: Done.

**Comment by korydraughn** [comment]
How do you feel about this being the commit message?
```
[_356] Improve error handling in load_oidc_configuration()
```

**Comment by MartinFlores751** [comment]
Updated.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by MartinFlores751** [comment]
Pounded.

**Pull request merged**